Assets/PlayTests/NuGetPlayTests.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections;
+using System.Runtime.InteropServices;
+using NugetForUnity;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+[assembly: PrebuildSetup(typeof(NuGetPlayTests))]
+[assembly: PostBuildCleanup(typeof(NuGetPlayTests))]
+
+/// <summary>
+/// Play mode tests allow us to install NuGet packages with Native code before play mode starts, then when play mode
+/// runs the native libraries are available for use.
+///
+/// There seems to be some Unity internals that prevents an edit mode test from adding a Native library and finding it
+/// in the same run. 
+/// </summary>
+public class NuGetPlayTests : IPrebuildSetup, IPostBuildCleanup
+{
+    readonly NugetPackageIdentifier sqlite = new NugetPackageIdentifier("SQLitePCLRaw.lib.e_sqlite3", "2.0.7");
+
+    /// <summary>
+    /// This is the version number of sqlite with periods replaced with zeros.
+    ///
+    /// Note: Version of the SQLite library does not match the NuGet package version 
+    /// </summary>
+    private readonly int _expectedVersion = 3035005;
+
+    [UnityTest]
+    public IEnumerator InstallAndRunSqlite()
+    {
+        yield return new WaitForFixedUpdate();
+
+        // Test the actual library by calling a "extern" method
+        var version = sqlite3_libversion_number();
+        Assert.That(version, Is.EqualTo(_expectedVersion));
+    }
+
+    /// <summary>
+    /// Call to the SQLite native file, this actually tests loading and access the library when called.
+    ///
+    /// On windows the call to sqlite3_libversion returns a garbled string so use the integer
+    /// </summary>
+    /// <returns></returns>
+    [DllImport("e_sqlite3")]
+    private static extern int sqlite3_libversion_number();
+
+    public void Setup()
+    {
+        try
+        {
+            sqlite3_libversion_number();
+            Assert.Fail("e_sqlite3 dll loaded, but should not be available");
+        }
+        catch (DllNotFoundException)
+        {
+        }
+
+        // For windows we end up importing two identical DLLs temporarily, this causes an error log that NUnit detects
+        // and would fail the test if we don't tell it to ignore the Failing messages
+        LogAssert.ignoreFailingMessages = true;
+        NugetHelper.InstallIdentifier(sqlite);
+        Assert.IsTrue(NugetHelper.IsInstalled(sqlite), "The package was NOT installed: {0} {1}", sqlite.Id,
+            sqlite.Version);
+    }
+
+    public void Cleanup()
+    {
+        // Need to reinitialise NugetHelper, but not sure why
+        NugetHelper.LoadNugetConfigFile();
+        NugetHelper.Uninstall(sqlite);
```

Comment:
I hit an exception here (if I fix the nuget config file exception) because it seems like the editor is still running. I think both of these issues may go away if you make this an edit mode test instead of a play mode one.

```
UnauthorizedAccessException: Access to the path "D:\Users\jowitt\source\External\NuGetForUnity\Assets\Packages\SQLitePCLRaw.lib.e_sqlite3.2.0.7\runtimes\win-x64\native\e_sqlite3.dll" is denied.
System.IO.File.Delete (System.String path) (at <eae584ce26bc40229c1b1aa476bfa589>:0)
System.IO.Directory.RecursiveDelete (System.String path) (at <eae584ce26bc40229c1b1aa476bfa589>:0)
System.IO.Directory.Delete (System.String path, System.Boolean recursive) (at <eae584ce26bc40229c1b1aa476bfa589>:0)
System.IO.DirectoryInfo.Delete (System.Boolean recursive) (at <eae584ce26bc40229c1b1aa476bfa589>:0)
(wrapper remoting-invoke-with-check) System.IO.DirectoryInfo.Delete(bool)
NugetForUnity.NugetHelper.DeleteDirectory (System.String directoryPath) (at Assets/NuGet/Editor/NugetHelper.cs:876)
NugetForUnity.NugetHelper.DeleteDirectory (System.String directoryPath) (at Assets/NuGet/Editor/NugetHelper.cs:862)
NugetForUnity.NugetHelper.DeleteDirectory (System.String directoryPath) (at Assets/NuGet/Editor/NugetHelper.cs:862)
NugetForUnity.NugetHelper.DeleteDirectory (System.String directoryPath) (at Assets/NuGet/Editor/NugetHelper.cs:862)
NugetForUnity.NugetHelper.Uninstall (NugetForUnity.NugetPackageIdentifier package, System.Boolean refreshAssets) (at Assets/NuGet/Editor/NugetHelper.cs:931)
NuGetPlayTests.Cleanup () (at Assets/PlayTests/NuGetPlayTests.cs:73)
UnityEditor.TestTools.TestRunner.TestLauncherBase+<>c.<ExecutePostBuildCleanupMethods>b__5_0 (UnityEngine.TestTools.IPostBuildCleanup targetClass) (at Library/PackageCache/com.unity.test-framework@1.1.29/UnityEditor.TestRunner/TestLaunchers/TestLauncherBase.cs:42)
UnityEditor.TestTools.TestRunner.TestLauncherBase.ExecuteMethods[T] (NUnit.Framework.Interfaces.ITest tests, NUnit.Framework.Interfaces.ITestFilter testRunnerFilter, UnityEditor.TestTools.TestRunner.AttributeFinderBase attributeFinder, System.String logString, System.Action`1[T] action, System.Nullable`1[T] testTargetPlatform) (at Library/PackageCache/com.unity.test-framework@1.1.29/UnityEditor.TestRunner/TestLaunchers/TestLauncherBase.cs:66)
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)

```

Reply:
Ah this is because you cannot delete a file in use on Windows, and Unity will not unload a native DLL once loaded :( 

We have to use play mode tests, we did try to use edit mode tests but we needed to force Unity to complete an asset database reload (or something like this) for it to pick up on the native files

Reply:
So this is caused by Windows not allowing you to delete a file that's in use, not sure why we didn't see it when testing on windows but we did see the need to restart Unity between tests.

I've added a catch around the deletes, that on a Windows machine will raise an error log message stating that native files cannot be deleted and a restart of Unity is needed. I'm not sure of any other solution to this I'm afraid